### PR TITLE
upower, power-profiles-daemon: Fix GI_TYPELIB_PATH for tests

### PR DESCRIPTION
**Pull request by YorikSar**
###### Description of changes

gobject-introspection doesn't handle checkInputs since https://github.com/NixOS/nixpkgs/pull/177538, so tests couldn't find UMockdev namespace. Added umockdev to buildInputs.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by YorikSar** [comment]
Note that this includes https://github.com/NixOS/nixpkgs/pull/182668 that fixes `umockdev` build in staging-next.

**Comment by YorikSar** [comment]
Looks like `power-profiles-daemon` also suffers from the same error. Will add fix to this branch.

**Comment by jtojnar** [comment]
Do not like it very much but I guess it is an okay workaround until it is fixed properly. cc @Artturin

**Comment by Artturin** [comment]
Im checking it out, do not merge yet

File: pkgs/os-specific/linux/power-profiles-daemon/default.nix [hunk 1]
```diff
@@ -67,6 +67,9 @@ stdenv.mkDerivation rec {
     glib
     polkit
     python3 # for cli tool
+    # Used only for check, but gobject-introspection doesn't populate
+    # GI_TYPELIB_PATH with checkInputs since https://github.com/NixOS/nixpkgs/pull/177538
```
Comment (Artturin) [inline pkgs/os-specific/linux/power-profiles-daemon/default.nix:71]: this is actually a manifestation of https://github.com/NixOS/nixpkgs/issues/161570 (checkInputs adds to nativeBuildInputs)
gobject-introspection actually does the right thing now so this issue was exposed

Comment (Artturin) [inline pkgs/os-specific/linux/power-profiles-daemon/default.nix:71]:
```suggestion
  # Move to check specific buildInputs once https://github.com/NixOS/nixpkgs/issues/161570 is solved
```

Comment (YorikSar) [reply pkgs/os-specific/linux/power-profiles-daemon/default.nix]: Reworded in both commits.

**Comment by Mindavi** [comment]
This broke cross-compilation for upower since umockdev doesn't cross-compile properly.

```
Installing /build/umockdev-0.17.13/src/umockdev-wrapper to /nix/store/1qblp8rylkg0a0y6jlfv8b2m8innkl9d-umockdev-aarch64-unknown-linux-gnu-0.17.13-bin/bin
Running custom install script '/nix/store/v35hli0b3na323sjh7q5057pxmqw2v5x-meson-0.61.2/bin/meson --internal gtkdoc --sourcedir=/build/umockdev-0.17.13 --builddir=/build/umockdev-0.17.13/build --subdir=docs/reference --headerdirs=/build/umockdev-0.17.13/build --mainfile=umockdev-docs.xml --modulename=umockdev --moduleversion= --mode=auto --gtkdoc-scan=/nix/store/n55ybhsaqp7alvpi7878wnqk2n6ry6xp-gtk-doc-1.33.2/bin/gtkdoc-scan --gtkdoc-scangobj=/nix/store/n55ybhsaqp7alvpi7878wnqk2n6ry6xp-gtk-doc-1.33.2/bin/gtkdoc-scangobj --gtkdoc-mkdb=/nix/store/n55ybhsaqp7alvpi7878wnqk2n6ry6xp-gtk-doc-1.33.2/bin/gtkdoc-mkdb --gtkdoc-mkhtml=/nix/store/n55ybhsaqp7alvpi7878wnqk2n6ry6xp-gtk-doc-1.33.2/bin/gtkdoc-mkhtml --gtkdoc-fixxref=/nix/store/n55ybhsaqp7alvpi7878wnqk2n6ry6xp-gtk-doc-1.33.2/bin/gtkdoc-fixxref --htmlargs= --scanargs=--ignore-decorators@@VALA_EXTERN --scanobjsargs= --gobjects-types-file=/build/umockdev-0.17.13/docs/reference/umockdev.types --fixxrefargs= --mkdbargs= --html-assets= --content-files=/build/umockdev-0.17.13/build/docs/reference/version.xml --expand-content-files= --ignore-headers=uevent_sender.h@@ioctl_tree.h@@debug.h --installdir= --cc=aarch64-unknown-linux-gnu-gcc --ld=aarch64-unknown-linux-gnu-gcc --cflags=-I/nix/store/jnrniljmhip62x6nycvcl3pllx55cd9l-glib-aarch64-unknown-linux-gnu-2.72.3-dev/include/glib-2.0 -I/nix/store/dfy1ajhqbz8ma60s3q9zqgirgyjv668r-glib-aarch64-unknown-linux-gnu-2.72.3/lib/glib-2.0/include -I/nix/store/jnrniljmhip62x6nycvcl3pllx55cd9l-glib-aarch64-unknown-linux-gnu-2.72.3-dev/include -I/nix/store/jnrniljmhip62x6nycvcl3pllx55cd9l-glib-aarch64-unknown-linux-gnu-2.72.3-dev/include/gio-unix-2.0 -I/nix/store/ynha2wg9nl4hrhys9il9j5vm72lmwxfj-libpcap-aarch64-unknown-linux-gnu-1.10.1/include -Werror=missing-prototypes -Werror=strict-prototypes -Werror=nested-externs -Werror=pointer-arith -Werror=implicit-function-declaration -Werror=pointer-arith -Werror=init-self -Werror=format-security -Werror=format=2 -Werror=unused-variable -Werror=return-type -Werror=uninitialized -Wcast-align -Wno-error=pragmas --ldflags=-L/build/umockdev-0.17.13/build/ -Wl,-rpath,/build/umockdev-0.17.13/build/ -lumockdev -L/nix/store/dfy1ajhqbz8ma60s3q9zqgirgyjv668r-glib-aarch64-unknown-linux-gnu-2.72.3/lib -lglib-2.0 -lgobject-2.0 -lgio-2.0 -ludev -lutil -L/nix/store/ynha2wg9nl4hrhys9il9j5vm72lmwxfj-libpcap-aarch64-unknown-linux-gnu-1.10.1/lib -lpcap'
--- stdout ---
Building documentation for umockdev
ERROR: Error in gtkdoc helper script:

ERROR: ['/nix/store/n55ybhsaqp7alvpi7878wnqk2n6ry6xp-gtk-doc-1.33.2/bin/gtkdoc-scangobj', '--types=/build/umockdev-0.17.13/docs/reference/umockdev.types', '--module=umockdev', '--run=', '--cflags=-I/nix/store/jnrniljmhip62x6nycvcl3pllx55cd9l-glib-aarch64-unknown-linux-gnu-2.72.3-dev/include/glib-2.0 -I/nix/store/dfy1ajhqbz8ma60s3q9zqgirgyjv668r-glib-aarch64-unknown-linux-gnu-2.72.3/lib/glib-2.0/include -I/nix/store/jnrniljmhip62x6nycvcl3pllx55cd9l-glib-aarch64-unknown-linux-gnu-2.72.3-dev/include -I/nix/store/jnrniljmhip62x6nycvcl3pllx55cd9l-glib-aarch64-unknown-linux-gnu-2.72.3-dev/include/gio-unix-2.0 -I/nix/store/ynha2wg9nl4hrhys9il9j5vm72lmwxfj-libpcap-aarch64-unknown-linux-gnu-1.10.1/include -Werror=missing-prototypes -Werror=strict-prototypes -Werror=nested-externs -Werror=pointer-arith -Werror=implicit-function-declaration -Werror=pointer-arith -Werror=init-self -Werror=format-security -Werror=format=2 -Werror=unused-variable -Werror=return-type -Werror=uninitialized -Wcast-align -Wno-error=pragmas', '--ldflags=-L/build/umockdev-0.17.13/build/ -Wl,-rpath,/build/umockdev-0.17.13/build/ -lumockdev -L/nix/store/dfy1ajhqbz8ma60s3q9zqgirgyjv668r-glib-aarch64-unknown-linux-gnu-2.72.3/lib -lglib-2.0 -lgobject-2.0 -lgio-2.0 -ludev -lutil -L/nix/store/ynha2wg9nl4hrhys9il9j5vm72lmwxfj-libpcap-aarch64-unknown-linux-gnu-1.10.1/lib -lpcap', '--cc=aarch64-unknown-linux-gnu-gcc', '--ld=aarch64-unknown-linux-gnu-gcc', '--output-dir=/build/umockdev-0.17.13/build/docs/reference'] failed with status 1
2022-07-24 22:29:19,374:scangobj.py:execute_command:1293:WARNING:Running scanner failed: [Errno 8] Exec format error: './umockdev-scan', command: ./umockdev-scan


--- stderr ---

FAILED: install script '/nix/store/v35hli0b3na323sjh7q5057pxmqw2v5x-meson-0.61.2/bin/meson --internal gtkdoc --sourcedir=/build/umockdev-0.17.13 --builddir=/build/umockdev-0.17.13/build --subdir=docs/reference --headerdirs=/build/umockdev-0.17.13/build --mainfile=umockdev-docs.xml --modulename=umockdev --moduleversion= --mode=auto --gtkdoc-scan=/nix/store/n55ybhsaqp7alvpi7878wnqk2n6ry6xp-gtk-doc-1.33.2/bin/gtkdoc-scan --gtkdoc-scangobj=/nix/store/n55ybhsaqp7alvpi7878wnqk2n6ry6xp-gtk-doc-1.33.2/bin/gtkdoc-scangobj --gtkdoc-mkdb=/nix/store/n55ybhsaqp7alvpi7878wnqk2n6ry6xp-gtk-doc-1.33.2/bin/gtkdoc-mkdb --gtkdoc-mkhtml=/nix/store/n55ybhsaqp7alvpi7878wnqk2n6ry6xp-gtk-doc-1.33.2/bin/gtkdoc-mkhtml --gtkdoc-fixxref=/nix/store/n55ybhsaqp7alvpi7878wnqk2n6ry6xp-gtk-doc-1.33.2/bin/gtkdoc-fixxref --htmlargs= --scanargs=--ignore-decorators@@VALA_EXTERN --scanobjsargs= --gobjects-types-file=/build/umockdev-0.17.13/docs/reference/umockdev.types --fixxrefargs= --mkdbargs= --html-assets= --content-files=/build/umockdev-0.17.13/build/docs/reference/version.xml --expand-content-files= --ignore-headers=uevent_sender.h@@ioctl_tree.h@@debug.h --installdir= --cc=aarch64-unknown-linux-gnu-gcc --ld=aarch64-unknown-linux-gnu-gcc --cflags=-I/nix/store/jnrniljmhip62x6nycvcl3pllx55cd9l-glib-aarch64-unknown-linux-gnu-2.72.3-dev/include/glib-2.0 -I/nix/store/dfy1ajhqbz8ma60s3q9zqgirgyjv668r-glib-aarch64-unknown-linux-gnu-2.72.3/lib/glib-2.0/include -I/nix/store/jnrniljmhip62x6nycvcl3pllx55cd9l-glib-aarch64-unknown-linux-gnu-2.72.3-dev/include -I/nix/store/jnrniljmhip62x6nycvcl3pllx55cd9l-glib-aarch64-unknown-linux-gnu-2.72.3-dev/include/gio-unix-2.0 -I/nix/store/ynha2wg9nl4hrhys9il9j5vm72lmwxfj-libpcap-aarch64-unknown-linux-gnu-1.10.1/include -Werror=missing-prototypes -Werror=strict-prototypes -Werror=nested-externs -Werror=pointer-arith -Werror=implicit-function-declaration -Werror=pointer-arith -Werror=init-self -Werror=format-security -Werror=format=2 -Werror=unused-variable -Werror=return-type -Werror=uninitialized -Wcast-align -Wno-error=pragmas --ldflags=-L/build/umockdev-0.17.13/build/ -Wl,-rpath,/build/umockdev-0.17.13/build/ -lumockdev -L/nix/store/dfy1ajhqbz8ma60s3q9zqgirgyjv668r-glib-aarch64-unknown-linux-gnu-2.72.3/lib -lglib-2.0 -lgobject-2.0 -lgio-2.0 -ludev -lutil -L/nix/store/ynha2wg9nl4hrhys9il9j5vm72lmwxfj-libpcap-aarch64-unknown-linux-gnu-1.10.1/lib -lpcap' exit code 1, stopped
FAILED: meson-install
/nix/store/v35hli0b3na323sjh7q5057pxmqw2v5x-meson-0.61.2/bin/meson install --no-rebuild
ninja: build stopped: subcommand failed.
```

I'll take a quick look to see if it's easily fixable.

**Comment by Artturin** [comment]
Proper fix is to use  `mesonEmulatorHook`

**Comment by Mindavi** [comment]
Oh! I thought it wasn't using meson, guess I'm a bit tired. Thanks.

**Comment by Artturin** [comment]
i'll send the fix to staging-next because I changed the logic of mesonEmulatorHook throw and if I add mesonEmulatorHook to it on master then when staging-next is merged there will be a throw

**Comment by Artturin** [comment]
https://github.com/nixos/nixpkgs/commit/12b2eaeeda9bb8e94a3257d5ce76a3604466fc29

**Pull request merged**